### PR TITLE
feat: Normal user list

### PR DESCRIPTION
**Pull request by jvillafanez**
## Description
Allow regular users to list other users.

People with account management permissions (such as admins) can list users as they have been until now. This PR won't affect them in any way.

Regular users (without account management permission) can list users with the following restrictions:
* Only "id", "displayName" and "mail" attributes will be shown per user.
* "$search" option MUST be present in the query, with at least 3 characters
* "$filter", "$apply", "$expand" and "$compute" options aren't allowed

## Related Issue
https://github.com/owncloud/ocis/issues/7782

## How Has This Been Tested?
Manually tested.
Unit tests have been adjusted because some of them where failing due to the user not having enough permissions

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: changelog/unreleased/enhancement-allow-listing-regular-users.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Enhancement: Allow regular users to list other users
+
+Regular users can search for other users, but having some limitations
```
Comment (JanAckermann) [inline changelog/unreleased/enhancement-allow-listing-regular-users.md:3]:
```suggestion
Regular users can search for other users, with some limitations.
```

Comment (JanAckermann) [reply changelog/unreleased/enhancement-allow-listing-regular-users.md:3]: Maybe list the limitations in the changelog item as it has high impact

**Review by JanAckermann — CHANGES_REQUESTED** [review]

**Review by JanAckermann — CHANGES_REQUESTED** [review]
Could you please add a test to validate that only the minimal set of id,displayName and mail is present if the regular user is requesting the users endpoint ?

File: changelog/unreleased/enhancement-allow-listing-regular-users.md (hunk 1 above)
Comment (rhafer) [reply changelog/unreleased/enhancement-allow-listing-regular-users.md]: done

**Comment by rhafer** [comment]
> Could you please add a test to validate that only the minimal set of id,displayName and mail is present if the regular user is requesting the users endpoint ?

@JanAckermann done. Please re-review

File: services/graph/pkg/service/v0/users.go [hunk 2]
```diff
@@ -207,6 +222,21 @@ func (g Graph) GetUsers(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	ctxHasFullPerms := g.contextUserHasFullAccountPerms(r.Context())
+	if !ctxHasFullPerms && (odataReq.Query == nil || odataReq.Query.Search == nil || len(odataReq.Query.Search.RawValue) < 3) {
```
Comment (micbar) [inline services/graph/pkg/service/v0/users.go:226]: Can we please make the minimum search length configurable?

**Review by micbar — APPROVED** [review]
One change request, rest looks fine.

**Comment by micbar** [comment]
Some getUser tests are failing.

**Review by micbar — APPROVED** [review]

**Comment by ScharfViktor** [comment]
>Only "id", "displayName" and "mail" attributes will be shown per user.

we can bug where we can get full info about user https://github.com/owncloud/ocis/issues/5125

**Pull request merged**